capstone/src/main/java/com/google/sps/servlets/StudentServlet.java

```diff
@@ -41,4 +50,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     response.sendRedirect("/profile.html");
   }
+
+  public ImmutableList<ImmutableList<String>> getAllAnnouncements(ImmutableList<String> clubNames) {
+    ImmutableList<ImmutableList<String>> announcements =
+        Streams.stream(clubNames)
+            .map(clubName -> PrototypeClubs.PROTOTYPE_CLUBS_MAP.get(clubName).getAnnouncements())
```

Comment:
You can do .flatMap() instead of .map() so it doesn't have to be a nested ImmutableList and can just be a single one.

Reply:
Done!